components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1132,26 +1135,38 @@ func socketActivationForDocker(ctx context.Context, wg *sync.WaitGroup, term *te
 	}
 
 	go func() {
-		<-ctx.Done()
+		<-newCtx.Done()
 		l.Close()
 	}()
 
 	_ = os.Chown(fn, gitpodUID, gitpodGID)
-	err = activation.Listen(ctx, l, func(socketFD *os.File) error {
-		cmd := exec.Command("docker-up")
+	err = activation.Listen(newCtx, l, func(socketFD *os.File) error {
+		cmd := exec.Command("/usr/bin/docker-up")
 		cmd.Env = append(os.Environ(), "LISTEN_FDS=1")
 		cmd.ExtraFiles = []*os.File{socketFD}
-		_, err := term.Start(cmd, terminal.TermOptions{
-			Annotations: map[string]string{
-				"supervisor": "true",
-			},
-			LogToStdout: true,
-		})
+		alias, err := term.Start(cmd, terminal.TermOptions{
+ 			Annotations: map[string]string{
+ 				"supervisor": "true",
+ 			},
+ 			LogToStdout: true,
+ 		})
+ 		if err != nil {
+ 			return err
+ 		}
+ 		pty, ok := term.Get(alias)
+ 		if !ok {
+ 			return errors.New("cannot find pty")
+ 		}
+ 		_, err = pty.Wait()
 		return err
 	})
 	if err != nil && !errors.Is(err, context.Canceled) {
 		log.WithError(err).Error("cannot provide Docker activation socket")
 	}
+ 	if !errors.Is(err, context.Canceled) {
+ 		wg.Add(1)
+ 		go socketActivationForDocker(ctx, wg, term)
+ 	}
```

Comment:
I'm wondering if, rather than making this recursive, it wouldn't be simpler to just wrap `activation.Listen` call in a for loop. WDYT?